sys/dev_vga.c

```diff
@@ -14,10 +14,54 @@ static int dev_vga_palette_write(vnode_t *v, uio_t *uio) {
   return vga_palette_write((vga_device_t *)v->v_data, uio);
 }
 
+#define RES_CTRL_BUFFER_SIZE 32
+
+static int dev_vga_resolution_write(vnode_t *v, uio_t *uio) {
+  vga_device_t *vga = (vga_device_t *)v->v_data;
+  uio->uio_offset = 0; /* This file does not support offsets. */
+  uint16_t xres, yres;
+  unsigned int bpp;
+  int error = vga_get_resolution(vga, &xres, &yres, &bpp);
+  if (error)
+    return error;
+  char buffer[RES_CTRL_BUFFER_SIZE];
+  error = uiomove_frombuf(buffer, RES_CTRL_BUFFER_SIZE, uio);
+  if (error)
+    return error;
+  /* Not specifying BPP leaves it at current value. */
+  int matches = sscanf(buffer, "%hd %hd %d", &xres, &yres, &bpp);
+  if (matches < 2)
+    return -EINVAL;
+  error = vga_set_resolution(vga, xres, yres, bpp);
+  if (error)
+    return error;
+  return 0;
+}
+
+static int dev_vga_resolution_read(vnode_t *v, uio_t *uio) {
+  vga_device_t *vga = (vga_device_t *)v->v_data;
+  uint16_t xres, yres;
+  unsigned int bpp;
+  int error = vga_get_resolution(vga, &xres, &yres, &bpp);
```

Comment:
Hmm... I think it's better to call it *video mode* rather than *resolution* because of `bpp` component. If choose to do so then control file should probably be named `videomode`